In [66]:
import pandas as pd
pd.options.mode.chained_assignment = None
import numpy as np
import matplotlib.pyplot as plt

In [67]:
house_data = pd.read_csv('../data/raw/Property_Assessment_Data__Historical_.csv')

c:\users\mitch\documents\data science side projects\edmonton property dashboard\prop_env\lib\site-packages\IPython\core\interactiveshell.py:3444: DtypeWarning: Columns (2,17,19) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [68]:
#current year as of oct 2021 -> accessed oct 27/2021
current_house_data = pd.read_csv('../data/raw/Property_Assessment_Data__Current_Calendar_Year_.csv')

In [69]:
current_house_data.head()

Account Number Suite  House Number          Street Name Garage  \
0        10177229   NaN           NaN                  NaN      N   
1         1194422   NaN  15613.000000         83 STREET NW      N   
2         1034222   NaN  10626.000000        161 AVENUE NW      Y   
3         1114768   NaN  14915.000000         76 AVENUE NW      Y   
4         1012285   NaN    564.000000  WOLF WILLOW ROAD NW      Y   

   Neighbourhood ID Neighbourhood                 Ward  Assessed Value  \
0       6213.000000    SUMMERSIDE         Karhiio Ward           52000   
1       2050.000000    BELLE RIVE  tastawiyiniwak Ward          242000   
2       3040.000000     BEAUMARIS  tastawiyiniwak Ward          411500   
3       4430.000000   RIO TERRACE   sipiwiyiniwak Ward          431500   
4       4610.000000     WESTRIDGE   sipiwiyiniwak Ward          686500   

   Latitude   Longitude                                  Point Location  \
0 53.420646 -113.475389    POINT (-113.4753890817257 53.42064580930636)   
1 53.616851 -113.467976   POINT (-113.46797578947897 53.61685142883479)   
2 53.620471 -113.507630  POINT (-113.50763003337708 53.620471257753714)   
3 53.508643 -113.579289  POINT (-113.57928868418604 53.508643477515854)   
4 53.507641 -113.603755  POINT (-113.60375458739135 53.507641466024275)   

   Assessment Class % 1  Assessment Class % 2  Assessment Class % 3  \
0                   100                   NaN                   NaN   
1                   100                   NaN                   NaN   
2                   100                   NaN                   NaN   
3                   100                   NaN                   NaN   
4                   100                   NaN                   NaN   

  Assessment Class 1 Assessment Class 2 Assessment Class 3  
0         COMMERCIAL                NaN                NaN  
1        RESIDENTIAL                NaN                NaN  
2        RESIDENTIAL                NaN                NaN  
3        RESIDENTIAL                NaN                NaN  
4        RESIDENTIAL                NaN                NaN

# House Data EDA

In [70]:
house_data.head()

Account Number  Assessment Year Suite  House Number       Street Name  \
0         1068451             2015   NaN   4007.000000  SAVARYN DRIVE SW   
1         1152594             2012   NaN  17310.000000      97 STREET NW   
2         1153568             2012   NaN           NaN               NaN   
3         1153626             2014   NaN           NaN               NaN   
4         1248590             2016   NaN           NaN               NaN   

  Legal Description  Latitude  Longitude Point Location       Neighbourhood  \
0    SW  22-51-24-4       NaN        NaN            NaN          SUMMERSIDE   
1               NaN       NaN        NaN            NaN            ELSINORE   
2               NaN       NaN        NaN            NaN  BATTERY OIL FIELDS   
3               NaN       NaN        NaN            NaN  BATTERY OIL FIELDS   
4     NW  3-52-25-4       NaN        NaN            NaN    TERWILLEGAR AREA   

   ...  Garage Zoning      Lot Size  Assessed Value  Assessment Class 1  \
0  ...       N    RPL 109314.321000  7597000.000000         RESIDENTIAL   
1  ...       N     AG      0.000000    12500.000000          COMMERCIAL   
2  ...       N     AG      0.000000    88000.000000          COMMERCIAL   
3  ...       N     AG      0.000000        0.000000          COMMERCIAL   
4  ...       N      A  10000.000000    49500.000000          COMMERCIAL   

  Assessment Class % 1  Assessment Class 2 Assessment Class % 2  \
0           100.000000                 NaN                  NaN   
1           100.000000                 NaN                  NaN   
2           100.000000                 NaN                  NaN   
3           100.000000                 NaN                  NaN   
4           100.000000                 NaN                  NaN   

   Assessment Class 3 Assessment Class % 3  
0                 NaN                  NaN  
1                 NaN                  NaN  
2                 NaN                  NaN  
3                 NaN                  NaN  
4                 NaN                  NaN  

[5 rows x 21 columns]

In [71]:
house_data.columns

Index(['Account Number', 'Assessment Year', 'Suite', 'House Number',
       'Street Name', 'Legal Description', 'Latitude', 'Longitude',
       'Point Location', 'Neighbourhood', 'Actual Year Built', 'Garage',
       'Zoning', 'Lot Size', 'Assessed Value', 'Assessment Class 1',
       'Assessment Class % 1', 'Assessment Class 2', 'Assessment Class % 2',
       'Assessment Class 3', 'Assessment Class % 3'],
      dtype='object')

In [72]:
house_data.shape

(3384820, 21)

In [73]:
house_data['Account Number'].is_unique

False

Some properties have multiple assessments

In [74]:
house_data['Assessment Class 1'].unique()

array(['RESIDENTIAL', 'COMMERCIAL', nan, 'FARMLAND', 'OTHER RESIDENTIAL',
       'NONRES MUNICIPAL/RES EDUCATION'], dtype=object)

In [75]:
house_data['Assessment Class 2'].unique()

array([nan, 'RESIDENTIAL', 'FARMLAND', 'COMMERCIAL', 'OTHER RESIDENTIAL',
       'NONRES MUNICIPAL/RES EDUCATION'], dtype=object)

In [76]:
house_data['Assessment Class 3'].unique()

array([nan, 'FARMLAND', 'COMMERCIAL', 'RESIDENTIAL', 'OTHER RESIDENTIAL',
       'NONRES MUNICIPAL/RES EDUCATION'], dtype=object)

In [77]:
house_data[~house_data['Assessment Class 2'].isna()]

Account Number  Assessment Year Suite  House Number  \
718             1008770             2015   NaN  10640.000000   
807             1001544             2013   NaN   2703.000000   
2272            1002294             2013   NaN   2304.000000   
3511            1002328             2019   NaN  18904.000000   
3543            1001361             2012   NaN   2005.000000   
...                 ...              ...   ...           ...   
3382038        10945385             2020   NaN    310.000000   
3382764        10945386             2020   NaN    420.000000   
3382790        10931838             2020   NaN  51012.000000   
3383112        10928855             2020   NaN   6415.000000   
3384610        10951325             2020   NaN  10615.000000   

                         Street Name  \
718                    180 STREET NW   
807                    184 STREET NW   
2272                   184 STREET NW   
3511                    23 AVENUE NW   
3543                   184 STREET NW   
...                              ...   
3382038              PARSONS ROAD SW   
3382764              PARSONS ROAD SW   
3382790  HIGHWAY 2 SERVICE ROAD 2 SW   
3383112                200 STREET SW   
3384610                 82 AVENUE NW   

                                         Legal Description  Latitude  \
718                                   Plan: 7214KS  Lot: 2 53.550800   
807                                          SE  4-52-25-4 53.459360   
2272                                         SE  5-52-25-4 53.457540   
3511                                         SE  5-52-25-4 53.457540   
3543                       NE  33-51-25-4 / NW  33-51-25-4 53.450810   
...                                                    ...       ...   
3382038                    Plan: 1922859  Block: 2  Lot: 2 53.431010   
3382764                    Plan: 1922859  Block: 2  Lot: 3 53.429930   
3382790                                                NaN 53.368920   
3383112                    Plan: 1920354  Block: 1  Lot: 5 53.374850   
3384610  Plan: I  Block: 64  Lot: 5 / Plan: I  Block: 6... 53.517770   

          Longitude               Point Location           Neighbourhood  ...  \
718     -113.634360   POINT (-113.63436 53.5508)       WILSON INDUSTRIAL  ...   
807     -113.622590  POINT (-113.62259 53.45936)              RURAL WEST  ...   
2272    -113.642990  POINT (-113.64299 53.45754)              RURAL WEST  ...   
3511    -113.649050  POINT (-113.64905 53.45754)             THE UPLANDS  ...   
3543    -113.634490  POINT (-113.63449 53.45081)              RURAL WEST  ...   
...             ...                          ...                     ...  ...   
3382038 -113.481450  POINT (-113.48145 53.43101)    ELLERSLIE INDUSTRIAL  ...   
3382764 -113.481490  POINT (-113.48149 53.42993)    ELLERSLIE INDUSTRIAL  ...   
3382790 -113.532450  POINT (-113.53245 53.36892)  EDMONTON SOUTH CENTRAL  ...   
3383112 -113.661250  POINT (-113.66125 53.37485)     EDMONTON SOUTH WEST  ...   
3384610 -113.504240  POINT (-113.50424 53.51777)         QUEEN ALEXANDRA  ...   

         Garage Zoning      Lot Size  Assessed Value  Assessment Class 1  \
718           N     IM  64683.449000  1779000.000000          COMMERCIAL   
807           N      A 212866.582000   552000.000000         RESIDENTIAL   
2272          N     AG 303777.470000    96500.000000         RESIDENTIAL   
3511          N     AG 303770.525000  2159000.000000         RESIDENTIAL   
3543          N     AG 383887.175000   760500.000000         RESIDENTIAL   
...         ...    ...           ...             ...                 ...   
3382038       N    EIB  21460.293000  2436500.000000          COMMERCIAL   
3382764       N    EIB  15775.722000  2436000.000000          COMMERCIAL   
3382790       Y    AES 156856.467000   660500.000000         RESIDENTIAL   
3383112       Y    AES  50029.309000   512500.000000         RESIDENTIAL   
3384610       N    DC1   3234.367000  7886500.000000   OTHER RESIDENTIAL   

        

Assessment class looks to classify if the property is a certain type of property (residential, commercial, etc.). Some properties can consist of multiple classes (eg. an apartment building with businesses on the main floor).

For this project, focused only on residential properties

In [78]:
resi_house_data = house_data[(house_data['Assessment Class 1'] == 'RESIDENTIAL') & (house_data['Assessment Class % 1'] == 100)]

In [79]:
resi_house_data

Account Number  Assessment Year Suite  House Number  \
0               1068451             2015   NaN   4007.000000   
7               1650860             2014   NaN           NaN   
9               2914836             2018   NaN  13004.000000   
14              3169893             2019   NaN   2503.000000   
28              9947689             2013   NaN     35.000000   
...                 ...              ...   ...           ...   
3384813        10937630             2020   NaN   3703.000000   
3384814        10961829             2020    43   1391.000000   
3384816        10954527             2020   NaN           NaN   
3384817        10954448             2020   116  10728.000000   
3384819        10943939             2020   NaN   5242.000000   

               Street Name                  Legal Description  Latitude  \
0         SAVARYN DRIVE SW                     SW  22-51-24-4       NaN   
7                      NaN   Plan: 3782RS  Block: 1  Lot: 30U 53.515450   
9            115 AVENUE NW    Plan: 1568HW  Block: 4  Lot: 21       NaN   
14           101 STREET SW            Plan: 8721140  Block: E       NaN   
28       ELSINORE PLACE NW                      NE  5-54-24-4       NaN   
...                    ...                                ...       ...   
3384813        2 STREET NW    Plan: 1922185  Block: 6  Lot: 4 53.472860   
3384814  STARLING DRIVE NW            Plan: 1621919  Unit: 56 53.598880   
3384816                NaN           Plan: 1923211  Unit: 87A 53.518410   
3384817       82 AVENUE NW            Plan: 1923211  Unit: 7A 53.518410   
3384819       23 AVENUE SW  Plan: 1922598  Block: 11  Lot: 33 53.412540   

          Longitude               Point Location         Neighbourhood  ...  \
0               NaN                          NaN            SUMMERSIDE  ...   
7       -113.616270  POINT (-113.61627 53.51545)            THORNCLIFF  ...   
9               NaN                          NaN             INGLEWOOD  ...   
14              NaN                          NaN  ELLERSLIE INDUSTRIAL  ...   
28              NaN                          NaN              ELSINORE  ...   
...             ...                          ...                   ...  ...   
3384813 -113.345290  POINT (-113.34529 53.47286)                 MAPLE  ...   
3384814 -113.652920  POINT (-113.65292 53.59888)              STARLING  ...   
3384816 -113.507100   POINT (-113.5071 53.51841)               GARNEAU  ...   
3384817 -113.507100   POINT (-113.5071 53.51841)               GARNEAU  ...   
3384819 -113.422800   POINT (-113.4228 53.41254)                WALKER  ...   

         Garage Zoning      Lot Size  Assessed Value  Assessment Class 1  \
0             N    RPL 109314.321000  7597000.000000         RESIDENTIAL   
7             N    RF1    118.072000        0.000000         RESIDENTIAL   
9             Y    RF3    490.277000   358500.000000         RESIDENTIAL   
14            N    AGI  49452.501000   244500.000000         RESIDENTIAL   
28            N    RF1    711.319000    55500.000000         RESIDENTIAL   
...         ...    ...           ...             ...                 ...   
3384813       N    RMD    334.741000    74500.000000         RESIDENTIAL   
3384814       Y    RA7    205.444000   281500.000000         RESIDENTIAL   
3384816       N    DC1      5.849000    10000.000000         RESIDENTIAL   
3384817       N    DC1     63.649000   408000.000000         RESIDENTIAL   
3384819       N    DC1    340.353000   146500.000000         RESIDENTIAL   

        Assessment Class % 1  Assessment Class 2 Assessment Class % 2  \
0                 100.000000                 NaN                  NaN   
7                 100.000000                 NaN                  NaN   
9                 100.000000                 NaN                  NaN   
14                100.000000                 NaN                  NaN   
28                100.000000                 NaN                  NaN   
...                      ...           

In [80]:
pd.set_option('float_format', '{:f}'.format)
resi_house_data['Assessed Value'].describe()

count    3159043.000000
mean      316251.180583
std       267990.034578
min            0.000000
25%       188500.000000
50%       315500.000000
75%       407500.000000
max     30081500.000000
Name: Assessed Value, dtype: float64

In [81]:
resi_house_data['Lot Size'].describe()

count   3133955.000000
mean        575.083918
std        6655.532823
min           0.000000
25%         116.873000
50%         386.330000
75%         568.710000
max      692660.676000
Name: Lot Size, dtype: float64

## Check if there are any nulls for assessed value or lot size 

In [82]:
resi_house_data[resi_house_data['Assessed Value'].isnull() == True]

Empty DataFrame
Columns: [Account Number, Assessment Year, Suite, House Number, Street Name, Legal Description, Latitude, Longitude, Point Location, Neighbourhood, Actual Year Built, Garage, Zoning, Lot Size, Assessed Value, Assessment Class 1, Assessment Class % 1, Assessment Class 2, Assessment Class % 2, Assessment Class 3, Assessment Class % 3]
Index: []

[0 rows x 21 columns]

In [83]:
resi_house_data[resi_house_data['Lot Size'].isnull() == True]

Account Number  Assessment Year Suite  House Number  \
40             10001923             2020   NaN   3442.000000   
43             10018932             2014   NaN    125.000000   
44             10018932             2015   NaN    125.000000   
45             10020385             2014   NaN      5.000000   
46             10020385             2019   NaN      5.000000   
...                 ...              ...   ...           ...   
3380215        10939888             2019   NaN   2003.000000   
3381478        10935242             2020   NaN    124.000000   
3382471        10930398             2019   NaN     58.000000   
3383050        10984049             2020   NaN    287.000000   
3383458        10953540             2020    31  12604.000000   

                Street Name Legal Description  Latitude   Longitude  \
40        LAKEVIEW POINT NW               NaN 53.552230 -113.696790   
43        RIDGEWAY DRIVE NW               NaN 53.503900 -113.359340   
44        RIDGEWAY DRIVE NW               NaN 53.503900 -113.359340   
45       QUARRY CRESCENT NW               NaN 53.499900 -113.362130   
46       QUARRY CRESCENT NW               NaN 53.499900 -113.362130   
...                     ...               ...       ...         ...   
3380215  WEST VIEW GREEN NW               NaN 53.552230 -113.696790   
3381478    OAK WOOD ROAD NW               NaN 53.504310 -113.363350   
3382471   EVERGREEN PARK NW               NaN 53.631220 -113.352220   
3383050   EVERGREEN PARK NW               NaN 53.631230 -113.354560   
3383458       153 AVENUE NW               NaN 53.618960 -113.539100   

                      Point Location                   Neighbourhood  ...  \
40       POINT (-113.69679 53.55223)                WESTVIEW VILLAGE  ...   
43        POINT (-113.35934 53.5039)                     MAPLE RIDGE  ...   
44        POINT (-113.35934 53.5039)                     MAPLE RIDGE  ...   
45        POINT (-113.36213 53.4999)                     MAPLE RIDGE  ...   
46        POINT (-113.36213 53.4999)                     MAPLE RIDGE  ...   
...                              ...                             ...  ...   
3380215  POINT (-113.69679 53.55223)                WESTVIEW VILLAGE  ...   
3381478  POINT (-113.36335 53.50431)  SOUTHEAST (ANNEXED) INDUSTRIAL  ...   
3382471  POINT (-113.35222 53.63122)                       EVERGREEN  ...   
3383050  POINT (-113.35456 53.63123)                       EVERGREEN  ...   
3383458   POINT (-113.5391 53.61896)                         DUNLUCE  ...   

         Garage Zoning Lot Size  Assessed Value  Assessment Class 1  \
40            Y    RMH      NaN   117500.000000         RESIDENTIAL   
43            N    RMH      NaN    43000.000000         RESIDENTIAL   
44            N    RMH      NaN    46500.000000         RESIDENTIAL   
45            N    RMH      NaN    71000.000000         RESIDENTIAL   
46            N    RMH      NaN    51000.000000         RESIDENTIAL   
...         ...    ...      ...             ...                 ...   
3380215       N    RMH      NaN    92500.000000         RESIDENTIAL   
3381478       N    RMH      NaN    60500.000000         RESIDENTIAL   
3382471       N    RMH      NaN    68000.000000         RESIDENTIAL   
3383050       N    RMH      NaN    97500.000000         RESIDENTIAL   
3383458       N    RMH      NaN   104500.000000         RESIDENTIAL   

        Assessment Class % 1  Assessment Class 2 Assessment Class % 2  \
40                100.000000                 NaN                  NaN   
43                100.000000                 NaN                  NaN   
44                100.000000                 NaN                  NaN   
45                100.000000                 NaN                  NaN   
46                100.000000                 NaN                  NaN   
...                      ...                 ...                  ...   
3380215           100.000000                 NaN                  NaN   
3381478           100.000

There are a number of assessed properties with no lot size info, lets take a closer look.

In [84]:
no_lot_size = resi_house_data[resi_house_data['Lot Size'].isnull() == True]
no_lot_size['Zoning'].value_counts()

RMH     20288
CR       3192
AG        729
IM        504
CNC       180
RA8h      128
IB         30
RF4        21
RF1         7
AGI         5
RR          2
RSL         1
DC1         1
Name: Zoning, dtype: int64

From zoning information, appears the vast majority of the lot size missing property assessments belong to mobile homes (RMH). CR zoning is not found on the Edmonton Zoning Bylaw, lets dig deeper.

In [85]:
cr_zone = no_lot_size[no_lot_size['Zoning'] == 'CR']
print(cr_zone['Street Name'].value_counts())
print(cr_zone['Neighbourhood'].value_counts())

WINTERBURN ROAD NW      3189
WEST STRAIT PLACE NW       3
Name: Street Name, dtype: int64
WESTVIEW VILLAGE                   2191
WINTERBURN INDUSTRIAL AREA WEST    1001
Name: Neighbourhood, dtype: int64


All the homes reside in the Winterburn area (nearby trailer park & industrial area). Will keep in mind when making predictions on lot size in this area.

## Check if every property has (address or legal description)

In [86]:
no_addy_df = resi_house_data[resi_house_data['Street Name'].isnull() & resi_house_data['Legal Description'].isnull()]
no_addy_df

Account Number  Assessment Year Suite  House Number Street Name  \
95923           1164698             2020   NaN           NaN         NaN   
97026           1164698             2016   NaN           NaN         NaN   
100883          1164698             2015   NaN           NaN         NaN   
102237          1164698             2019   NaN           NaN         NaN   
105983          1164698             2018   NaN           NaN         NaN   
...                 ...              ...   ...           ...         ...   
3381508        10936105             2020   NaN           NaN         NaN   
3383629        10971296             2020   NaN           NaN         NaN   
3384028        10963810             2020   NaN           NaN         NaN   
3384123        10936091             2019   NaN           NaN         NaN   
3384812        10935148             2020   NaN           NaN         NaN   

        Legal Description  Latitude   Longitude               Point Location  \
95923                 NaN 53.577990 -113.317000    POINT (-113.317 53.57799)   
97026                 NaN 53.577990 -113.317000    POINT (-113.317 53.57799)   
100883                NaN 53.577990 -113.317000    POINT (-113.317 53.57799)   
102237                NaN 53.577990 -113.317000    POINT (-113.317 53.57799)   
105983                NaN 53.577990 -113.317000    POINT (-113.317 53.57799)   
...                   ...       ...         ...                          ...   
3381508               NaN 53.501280 -113.652170  POINT (-113.65217 53.50128)   
3383629               NaN 53.607920 -113.393990  POINT (-113.39399 53.60792)   
3384028               NaN 53.513520 -113.643810  POINT (-113.64381 53.51352)   
3384123               NaN 53.507800 -113.468050   POINT (-113.46805 53.5078)   
3384812               NaN 53.455550 -113.464230  POINT (-113.46423 53.45555)   

           Neighbourhood  ...  Garage Zoning     Lot Size  Assessed Value  \
95923    CLOVER BAR AREA  ...       N     IH 16882.787000    83500.000000   
97026    CLOVER BAR AREA  ...       N     IH 16882.787000    83500.000000   
100883   CLOVER BAR AREA  ...       N     IH 16882.787000    83500.000000   
102237   CLOVER BAR AREA  ...       N     IH 16882.787000    83500.000000   
105983   CLOVER BAR AREA  ...       N     IH 16882.787000    83500.000000   
...                  ...  ...     ...    ...          ...             ...   
3381508     ORMSBY PLACE  ...       N    RF1   202.957000     7500.000000   
3383629         KIRKNESS  ...       N    RF4   446.751000    16500.000000   
3384028       ALDERGROVE  ...       N    RF1   132.700800     4500.000000   
3384123        HAZELDEAN  ...       N    RF3   241.957000     9000.000000   
3384812         MEYONOHK  ...       N    RF1   157.130000     5500.000000   

         Assessment Class 1 Assessment Class % 1  Assessment Class 2  \
95923           RESIDENTIAL           100.000000                 NaN   
97026           RESIDENTIAL           100.000000                 NaN   
100883          RESIDENTIAL           100.000000                 NaN   
102237          RESIDENTIAL           100.000000                 NaN   
105983          RESIDENTIAL           100.000000                 NaN   
...                     ...                  ...                 ...   
3381508         RESIDENTIAL           100.000000                 NaN   
3383629         RESIDENTIAL           100.000000                 NaN   
3384028         RESIDENTIAL           100.000000                 NaN   
3384123         RESIDENTIAL           100.000000                 NaN   
3384812         RESIDENTIAL           100.000000                 NaN   

        Assessment Class % 2  Assessment Class 3 Assessment Class % 3  
95923                    NaN                 NaN                  NaN  
97026                    NaN                 NaN                  NaN  
100883                   NaN                 NaN                  NaN  
102237                   NaN                 NaN         

In [87]:
no_addy_df['Account Number'].value_counts()

9994733     9
9992204     9
9995012     9
9994708     9
9995505     9
           ..
10965549    1
10937466    1
10095525    1
10191526    1
10935148    1
Name: Account Number, Length: 402, dtype: int64

There are 402 properties with no address - lets confirm that they all have point locations at least.

In [88]:
no_addy_df[no_addy_df['Point Location'].isnull() == True]

Account Number  Assessment Year Suite  House Number Street Name  \
1996073         9988893             2013   NaN           NaN         NaN   
1998906         9988893             2016   NaN           NaN         NaN   
2002300         9988893             2017   NaN           NaN         NaN   
2004027         9988893             2015   NaN           NaN         NaN   
2007441         9988893             2012   NaN           NaN         NaN   
2009414         9988893             2014   NaN           NaN         NaN   

        Legal Description  Latitude  Longitude Point Location Neighbourhood  \
1996073               NaN       NaN        NaN            NaN     WESTMOUNT   
1998906               NaN       NaN        NaN            NaN     WESTMOUNT   
2002300               NaN       NaN        NaN            NaN     WESTMOUNT   
2004027               NaN       NaN        NaN            NaN     WESTMOUNT   
2007441               NaN       NaN        NaN            NaN     WESTMOUNT   
2009414               NaN       NaN        NaN            NaN     WESTMOUNT   

         ...  Garage Zoning   Lot Size  Assessed Value  Assessment Class 1  \
1996073  ...       N    RF3 208.807000     7500.000000         RESIDENTIAL   
1998906  ...       N    RF3 208.807000     7500.000000         RESIDENTIAL   
2002300  ...       N    RF3 208.807000     7500.000000         RESIDENTIAL   
2004027  ...       N    RF3 208.807000     7500.000000         RESIDENTIAL   
2007441  ...       N    RF3 208.807000     6000.000000         RESIDENTIAL   
2009414  ...       N    RF3 208.807000     7500.000000         RESIDENTIAL   

        Assessment Class % 1  Assessment Class 2 Assessment Class % 2  \
1996073           100.000000                 NaN                  NaN   
1998906           100.000000                 NaN                  NaN   
2002300           100.000000                 NaN                  NaN   
2004027           100.000000                 NaN                  NaN   
2007441           100.000000                 NaN                  NaN   
2009414           100.000000                 NaN                  NaN   

         Assessment Class 3 Assessment Class % 3  
1996073                 NaN                  NaN  
1998906                 NaN                  NaN  
2002300                 NaN                  NaN  
2004027                 NaN                  NaN  
2007441                 NaN                  NaN  
2009414                 NaN                  NaN  

[6 rows x 21 columns]

Here is one property with no address or map location. Will remove this property from dataset.

In [89]:
remove_index_list = no_addy_df[no_addy_df['Point Location'].isnull() == True].index

In [90]:
print(resi_house_data.shape)
resi_house_data.drop(remove_index_list, inplace=True)
#confirm drop
print(resi_house_data.shape)

(3159043, 21)
(3159037, 21)


## Year to Year Stats

In [91]:
resi_house_data['Assessment Year'].unique()

array([2015, 2014, 2018, 2019, 2013, 2016, 2020, 2017, 2012], dtype=int64)

Assessment years range from 2012-2020

In [92]:
year_list = resi_house_data['Assessment Year'].unique()
year_list = np.sort(year_list)

In [93]:
def stat_per_year(df, stat_column_name, year_list):
    stat_avg_list = []
    for year in year_list:
        year_mask = df[df['Assessment Year'] == year]
        stat_avg = year_mask[stat_column_name].mean()
        stat_avg_list.append(stat_avg)
    return stat_avg_list

In [94]:
stat_per_year(resi_house_data, 'Assessed Value', year_list)

[297939.73504852323,
 303020.2066663972,
 310057.6833790588,
 328292.46505910385,
 329222.0068799305,
 320403.90860732226,
 320276.01980874134,
 321294.3759825549,
 312000.6391176764]

In [95]:
stat_per_year(resi_house_data, 'Lot Size', year_list)

[498.58174454437903,
 505.429239769213,
 645.5308083319143,
 640.7443754288332,
 617.4191135074295,
 601.7628024422843,
 576.3431685178066,
 552.4317688838518,
 532.055321387008]

### incorporate wards into dataset

In [96]:
current_house_data['Ward'].value_counts()

O-day'min Ward                38310
Ipiihkoohkanipiaohtsi Ward    36914
Métis Ward                    36510
pihêsiwin Ward                36353
sipiwiyiniwak Ward            36152
Karhiio Ward                  35373
Nakota Isga Ward              34355
Dene Ward                     33862
Anirniq Ward                  33571
papastew Ward                 31793
tastawiyiniwak Ward           29961
Sspomitapi Ward               28006
Name: Ward, dtype: int64

In [97]:
ward_mask = current_house_data[['Ward','Neighbourhood']]
ward_mask

Ward   Neighbourhood
0                     Karhiio Ward      SUMMERSIDE
1              tastawiyiniwak Ward      BELLE RIVE
2              tastawiyiniwak Ward       BEAUMARIS
3               sipiwiyiniwak Ward     RIO TERRACE
4               sipiwiyiniwak Ward       WESTRIDGE
...                            ...             ...
411161  Ipiihkoohkanipiaohtsi Ward  CHAPPELLE AREA
411162  Ipiihkoohkanipiaohtsi Ward        CAVANAGH
411163            Nakota Isga Ward   WEBBER GREENS
411164          sipiwiyiniwak Ward        EDGEMONT
411165            Nakota Isga Ward        STARLING

[411166 rows x 2 columns]

In [98]:
ward_mask.groupby('Ward')['Neighbourhood'].unique()

Ward
Anirniq Ward                  [HAGMANN ESTATE INDUSTRIAL, GAGNON ESTATE INDU...
Dene Ward                     [RURAL NORTH EAST HORSE HILL, FRASER, BANNERMA...
Ipiihkoohkanipiaohtsi Ward    [SKYRATTLER, KEHEEWIN, BEARSPAW, BLUE QUILL, R...
Karhiio Ward                  [SUMMERSIDE, MEYONOHK, SAKAW, ELLERSLIE, MEYOK...
Métis Ward                    [EASTGATE BUSINESS PARK, WEIR INDUSTRIAL, MONT...
Nakota Isga Ward              [GLENWOOD, WEST JASPER PLACE, ALDERGROVE, WINT...
O-day'min Ward                [OLIVER, CENTRAL MCDOUGALL, WESTMOUNT, QUEEN M...
Sspomitapi Ward               [MAPLE RIDGE, BISSET, WEINLOS, POLLARD MEADOWS...
papastew Ward                 [STRATHCONA, QUEEN ALEXANDRA, GARNEAU, EMPIRE ...
pihêsiwin Ward                [RAMSAY HEIGHTS, HENDERSON ESTATES, RHATIGAN R...
sipiwiyiniwak Ward            [RIO TERRACE, WESTRIDGE, MEADOWLARK PARK, JASP...
tastawiyiniwak Ward           [BELLE RIVE, BEAUMARIS, LAGO LINDO, KILKENNY, ...
Name: Neighbourhood, dtype: object

In [99]:
ward_dict = ward_mask.groupby('Ward')['Neighbourhood'].unique().to_dict()

In [100]:
ward_dict = {k:v.tolist() for k,v in ward_dict.items()}

In [101]:
ward_dict

{'Anirniq Ward': ['HAGMANN ESTATE INDUSTRIAL',
  'GAGNON ESTATE INDUSTRIAL',
  'MISTATIM INDUSTRIAL',
  'DUNLUCE',
  'LAUDERDALE',
  'CARLISLE',
  'PRINCE CHARLES',
  'MITCHELL INDUSTRIAL',
  'DOVERCOURT',
  'GOODRIDGE CORNERS',
  'INGLEWOOD',
  'CARLTON',
  'ROSSLYN',
  'RAMPART INDUSTRIAL',
  'HIGH PARK INDUSTRIAL',
  'ANTHONY HENDAY MISTATIM',
  'CALDER',
  'GARSIDE INDUSTRIAL',
  'CUMBERLAND',
  'ANTHONY HENDAY RAMPART',
  'PEMBINA',
  'KENSINGTON',
  'HUFF BREMNER ESTATE INDUSTRIAL',
  'CANOSSA',
  'RAPPERSWILL',
  'WOODCROFT',
  'BONAVENTURE INDUSTRIAL',
  'BROWN INDUSTRIAL',
  'ATHLONE',
  'YELLOWHEAD CORRIDOR WEST',
  'BARANOW',
  'SHERBROOKE',
  'DOMINION INDUSTRIAL',
  'WELLINGTON',
  'MCARTHUR INDUSTRIAL',
  'ALBANY',
  'CAERNARVON',
  'OXFORD',
  'GRIESBACH',
  'HUDSON'],
 'Dene Ward': ['RURAL NORTH EAST HORSE HILL',
  'FRASER',
  'BANNERMAN',
  'HAIRSINE',
  'BELVEDERE',
  'CLOVER BAR AREA',
  'CASSELMAN',
  'EDMONTON ENERGY AND TECHNOLOGY PARK',
  'RURAL NORTH EAST SOUTH 

In [102]:
ward_dict['Dene Ward']

['RURAL NORTH EAST HORSE HILL',
 'FRASER',
 'BANNERMAN',
 'HAIRSINE',
 'BELVEDERE',
 'CLOVER BAR AREA',
 'CASSELMAN',
 'EDMONTON ENERGY AND TECHNOLOGY PARK',
 'RURAL NORTH EAST SOUTH STURGEON',
 'KIRKNESS',
 'ANTHONY HENDAY HORSE HILL',
 'YORK',
 'EBBERS',
 'HOMESTEADER',
 'MARQUIS',
 'GORMAN',
 'KENNEDALE INDUSTRIAL',
 'CY BECKER',
 'MCCONACHIE AREA',
 'BRINTNELL',
 'ANTHONY HENDAY ENERGY PARK',
 'OVERLANDERS',
 'EVERGREEN',
 'ANTHONY HENDAY CLAREVIEW',
 'MCLEOD',
 'MATT BERRY',
 'HOLLICK-KENYON',
 'KERNOHAN',
 'RIVER VALLEY HERMITAGE',
 'CLAREVIEW TOWN CENTRE',
 'BELMONT',
 'CANON RIDGE',
 'SIFTON PARK',
 'MILLER']

In [103]:
test = 'BEARSPAW'
for index, value in enumerate(ward_dict.values()):
    if test in value:   
        print(list(ward_dict)[index])
    else:
        print('no')

no
no
Ipiihkoohkanipiaohtsi Ward
no
no
no
no
no
no
no
no
no


In [104]:
list(ward_dict)[2]

'Ipiihkoohkanipiaohtsi Ward'

In [105]:
def nbhd_fill(row):
    for index, value in enumerate(ward_dict.values()):
        if row['Neighbourhood'] in value:
#             print(list(ward_dict)[index])
#             print(row['Ward'])
            row['Ward'] = list(ward_dict)[index]
    return row['Ward']


In [106]:
resi_house_data['Ward'] = 0
resi_house_data.head()

Account Number  Assessment Year Suite  House Number        Street Name  \
0          1068451             2015   NaN   4007.000000   SAVARYN DRIVE SW   
7          1650860             2014   NaN           NaN                NaN   
9          2914836             2018   NaN  13004.000000      115 AVENUE NW   
14         3169893             2019   NaN   2503.000000      101 STREET SW   
28         9947689             2013   NaN     35.000000  ELSINORE PLACE NW   

                   Legal Description  Latitude   Longitude  \
0                     SW  22-51-24-4       NaN         NaN   
7   Plan: 3782RS  Block: 1  Lot: 30U 53.515450 -113.616270   
9    Plan: 1568HW  Block: 4  Lot: 21       NaN         NaN   
14           Plan: 8721140  Block: E       NaN         NaN   
28                     NE  5-54-24-4       NaN         NaN   

                 Point Location         Neighbourhood  ...  Zoning  \
0                           NaN            SUMMERSIDE  ...     RPL   
7   POINT (-113.61627 53.51545)            THORNCLIFF  ...     RF1   
9                           NaN             INGLEWOOD  ...     RF3   
14                          NaN  ELLERSLIE INDUSTRIAL  ...     AGI   
28                          NaN              ELSINORE  ...     RF1   

        Lot Size Assessed Value  Assessment Class 1  Assessment Class % 1  \
0  109314.321000 7597000.000000         RESIDENTIAL            100.000000   
7     118.072000       0.000000         RESIDENTIAL            100.000000   
9     490.277000  358500.000000         RESIDENTIAL            100.000000   
14  49452.501000  244500.000000         RESIDENTIAL            100.000000   
28    711.319000   55500.000000         RESIDENTIAL            100.000000   

   Assessment Class 2  Assessment Class % 2 Assessment Class 3  \
0                 NaN                   NaN                NaN   
7                 NaN                   NaN                NaN   
9                 NaN                   NaN                NaN   
14                NaN                   NaN                NaN   
28                NaN                   NaN                NaN   

    Assessment Class % 3 Ward  
0                    NaN    0  
7                    NaN    0  
9                    NaN    0  
14                   NaN    0  
28                   NaN    0  

[5 rows x 22 columns]

In [107]:
nbhd_fill(resi_house_data.iloc[0,:])

'Karhiio Ward'

In [108]:
resi_house_data['Ward'].iloc[0,:] = 1
resi_house_data.iloc[0,:]['Ward']

1

In [109]:
resi_house_data['Ward'] = 0
resi_house_data['Ward'] = resi_house_data.apply(nbhd_fill, axis=1)

In [110]:
resi_house_data['Ward'].value_counts()

O-day'min Ward                294912
sipiwiyiniwak Ward            293127
Métis Ward                    292853
pihêsiwin Ward                265399
Dene Ward                     263119
Ipiihkoohkanipiaohtsi Ward    258105
Karhiio Ward                  255839
Anirniq Ward                  251399
papastew Ward                 250785
tastawiyiniwak Ward           249039
Nakota Isga Ward              247808
Sspomitapi Ward               213862
0                              22790
Name: Ward, dtype: int64

Still some missing wards in the ward column, lets check out

In [111]:
no_ward_mask = resi_house_data[resi_house_data['Ward'] == 0]

In [112]:
no_wards_list = no_ward_mask['Neighbourhood'].unique().tolist()
no_wards_list

['SOUTHEAST (ANNEXED) INDUSTRIAL',
 'RURAL WEST',
 'HERITAGE VALLEY AREA EAST',
 'RURAL SOUTH EAST',
 'MEADOWS AREA',
 'RIVER VALLEY WINDEMERE',
 'GLENRIDDING AREA',
 'MAGRATH HEIGHTS AREA',
 'RURAL NORTH EAST NORTH STURGEON',
 'RURAL NORTH WEST',
 'RAPPERSWIL',
 'RURAL WEST BIG LAKE',
 'RURAL WEST LEWIS FARMS',
 'ANTHONY HENDAY SOUTHEAST',
 'PLACE LA RUE',
 'CRYSTALLINA NERA',
 'WESTBROOK ESTATE',
 'PILOT SOUND AREA WEST PORTION',
 'LAKE DISTRICT NORTH EAST PORTION',
 'CPR WEST',
 'MILL CREEK RAVINE',
 'MACTAGGART AREA',
 'TERWILLEGAR SOUTH']

Rapperswil should be in Anirniq Ward?

In [113]:
ward_dict['Anirniq Ward']

['HAGMANN ESTATE INDUSTRIAL',
 'GAGNON ESTATE INDUSTRIAL',
 'MISTATIM INDUSTRIAL',
 'DUNLUCE',
 'LAUDERDALE',
 'CARLISLE',
 'PRINCE CHARLES',
 'MITCHELL INDUSTRIAL',
 'DOVERCOURT',
 'GOODRIDGE CORNERS',
 'INGLEWOOD',
 'CARLTON',
 'ROSSLYN',
 'RAMPART INDUSTRIAL',
 'HIGH PARK INDUSTRIAL',
 'ANTHONY HENDAY MISTATIM',
 'CALDER',
 'GARSIDE INDUSTRIAL',
 'CUMBERLAND',
 'ANTHONY HENDAY RAMPART',
 'PEMBINA',
 'KENSINGTON',
 'HUFF BREMNER ESTATE INDUSTRIAL',
 'CANOSSA',
 'RAPPERSWILL',
 'WOODCROFT',
 'BONAVENTURE INDUSTRIAL',
 'BROWN INDUSTRIAL',
 'ATHLONE',
 'YELLOWHEAD CORRIDOR WEST',
 'BARANOW',
 'SHERBROOKE',
 'DOMINION INDUSTRIAL',
 'WELLINGTON',
 'MCARTHUR INDUSTRIAL',
 'ALBANY',
 'CAERNARVON',
 'OXFORD',
 'GRIESBACH',
 'HUDSON']

Rapperswil is misspelt, should be Rapperswill. This continues with other neighbourhoods as well, many neighbourhoods have word "area" with it, causing the dictionary map to not work. Lets continue cleaning.

In [114]:
resi_house_data['Neighbourhood'].replace('RAPPERSWIL', 'RAPPERSWILL', inplace=True)
resi_house_data[resi_house_data['Neighbourhood'] == 'RAPPERSWIL']
no_wards_list.remove('RAPPERSWIL')

In [115]:
no_wards_list.sort()
no_wards_list

['ANTHONY HENDAY SOUTHEAST',
 'CPR WEST',
 'CRYSTALLINA NERA',
 'GLENRIDDING AREA',
 'HERITAGE VALLEY AREA EAST',
 'LAKE DISTRICT NORTH EAST PORTION',
 'MACTAGGART AREA',
 'MAGRATH HEIGHTS AREA',
 'MEADOWS AREA',
 'MILL CREEK RAVINE',
 'PILOT SOUND AREA WEST PORTION',
 'PLACE LA RUE',
 'RIVER VALLEY WINDEMERE',
 'RURAL NORTH EAST NORTH STURGEON',
 'RURAL NORTH WEST',
 'RURAL SOUTH EAST',
 'RURAL WEST',
 'RURAL WEST BIG LAKE',
 'RURAL WEST LEWIS FARMS',
 'SOUTHEAST (ANNEXED) INDUSTRIAL',
 'TERWILLEGAR SOUTH',
 'WESTBROOK ESTATE']

- 'ANTHONY HENDAY SOUTHEAST' belongs in Karhiio, 
- 'CPR WEST' now known as STRATHCONA JUNCTION in papastew,
- 'CRYSTALLINA NERA' belongs in tastawiyiniwak,
- 'GLENRIDDING AREA' belongs in pihêsiwin,
- 'HERITAGE VALLEY AREA EAST' belongs in Ipiihkoohkanipiaohtsi,
- 'LAKE DISTRICT NORTH EAST PORTION' belongs in tastawiyiniwak,
- 'MACTAGGART AREA' belongs in pihêsiwin,
- 'MAGRATH HEIGHTS AREA' belongs in pihêsiwin,
- 'MEADOWS AREA' belongs in Sspomitapi,
- 'MILL CREEK RAVINE' belongs in papastew,
- 'PILOT SOUND AREA WEST PORTION' belongs in Dene,
- 'PLACE LA RUE' should be 'PLACE LARUE',
- 'RIVER VALLEY WINDEMERE' is misspelt should be 'RIVER VALLEY WINDERMERE'
- 'RURAL NORTH EAST NORTH STURGEON' belongs in Dene,
- 'RURAL NORTH WEST' belongs in Anirniq,
- 'RURAL SOUTH EAST' belongs in Sspomitapi,
- 'RURAL WEST' belongs in sipiwiyiniwak,
- 'RURAL WEST BIG LAKE' belongs in Nakota Isga,
- 'RURAL WEST LEWIS FARMS' belongs in Nakota Isga,
- 'SOUTHEAST (ANNEXED) INDUSTRIAL' belongs in Sspomitapi,
- 'TERWILLEGAR SOUTH' should be 'SOUTH TERWILLEGAR',
- 'WESTBROOK ESTATE' should be 'WESTBROOK ESTATES'


In [116]:
def ward_assign(neigh_name, ward_name):
    resi_house_data['Ward'].loc[resi_house_data['Neighbourhood'] == neigh_name] = ward_name

In [117]:
def rename_neigh(incorrect_name, replace_name):
    resi_house_data['Neighbourhood'].replace(incorrect_name, replace_name, inplace=True)
    no_wards_list.remove(incorrect_name)

In [118]:
ward_assign('ANTHONY HENDAY SOUTHEAST', 'Karhiio')
ward_assign('CRYSTALLINA NERA', 'tastawiyiniwak')
ward_assign('GLENRIDDING AREA', 'pihêsiwin')
ward_assign('HERITAGE VALLEY AREA EAST', 'Ipiihkoohkanipiaohtsi')
ward_assign('LAKE DISTRICT NORTH EAST PORTION', 'tastawiyiniwak')
ward_assign('MACTAGGART AREA', 'pihêsiwin')
ward_assign('MAGRATH HEIGHTS AREA', 'pihêsiwin')
ward_assign('MEADOWS AREA', 'Sspomitapi')
ward_assign('MILL CREEK RAVINE', 'papastew')
ward_assign('PILOT SOUND AREA WEST PORTION', 'Dene')
ward_assign('RURAL NORTH EAST NORTH STURGEON', 'Dene')
ward_assign('RURAL NORTH WEST', 'Anirniq')
ward_assign('RURAL SOUTH EAST', 'Sspomitapi')
ward_assign('RURAL WEST', 'sipiwiyiniwak')
ward_assign('RURAL WEST BIG LAKE', 'Nakota Isga')
ward_assign('RURAL WEST LEWIS FARMS', 'Nakota Isga')
ward_assign('SOUTHEAST (ANNEXED) INDUSTRIAL', 'Sspomitapi')

In [119]:
rename_neigh('CPR WEST', 'STRATHCONA JUNCTION')
rename_neigh('PLACE LA RUE', 'PLACE LARUE')
rename_neigh('RIVER VALLEY WINDEMERE', 'RIVER VALLEY WINDERMERE')
rename_neigh('TERWILLEGAR SOUTH', 'SOUTH TERWILLEGAR')
rename_neigh('WESTBROOK ESTATE', 'WESTBROOK ESTATES')

In [120]:
resi_house_data['Ward'] = resi_house_data.apply(nbhd_fill, axis=1)

Check to see if all Wards are now labelled

In [122]:
no_ward_mask = resi_house_data[resi_house_data['Ward'] == 0]
no_ward_mask

Empty DataFrame
Columns: [Account Number, Assessment Year, Suite, House Number, Street Name, Legal Description, Latitude, Longitude, Point Location, Neighbourhood, Actual Year Built, Garage, Zoning, Lot Size, Assessed Value, Assessment Class 1, Assessment Class % 1, Assessment Class 2, Assessment Class % 2, Assessment Class 3, Assessment Class % 3, Ward]
Index: []

[0 rows x 22 columns]

Wards are now all labelled!